featuretools/entityset/entityset.py

```diff
@@ -319,24 +319,8 @@ def add_relationship(self,
         return self
 
     def set_secondary_time_index(self, entity, secondary_time_index):
-        for time_index, columns in secondary_time_index.items():
-            if is_instance(entity.df, (dd, ks), 'DataFrame') or entity.df.empty:
-                variable_dtype = entity[time_index]._default_pandas_dtype
-                time_to_check = vtypes.DEFAULT_DTYPE_VALUES[variable_dtype]
-            else:
-                time_to_check = entity.df[time_index].head(1).iloc[0]
-            time_type = _check_time_type(time_to_check)
-            if time_type is None:
-                raise TypeError("%s time index not recognized as numeric or"
-                                " datetime" % (entity.id))
-            if self.time_type != time_type:
-                raise TypeError("%s time index is %s type which differs from"
-                                " other entityset time indexes" %
-                                (entity.id, time_type))
-            if time_index not in columns:
-                columns.append(time_index)
-
         entity.secondary_time_index = secondary_time_index
+        self._check_secondary_time_index(entity)
```

Comment:
Don't we want to validate before setting the secondary time index?  Even if `_check_secondary_time_index` reports an error, we'll still have set a bad secondary time index

Reply:
yeah, makes sense

✅  88fb837d